python_modules/dagster/dagster/_core/types/builtin_config_schemas.py

```diff
@@ -32,6 +33,15 @@ def define_typed_input_schema_dict(value_config_type):
 
 
 def load_type_input_schema_dict(value):
+    input_schema_dict_format_str = "{'value'|'json'|'pickle': <actual_value>}"
+    if not isinstance(value, collections.Mapping):
+        check.failed(
+            f"Expected input schema dict, but instead received a value of type {type(value)}. An input schema dict is of the format {input_schema_dict_format_str}"
+        )
+    if len(value) > 1 or ("value" not in value and "json" not in value and "pickle" not in value):
+        check.failed(
```

Comment:
is `check` the right type of error to throw here? How do these errors show up in `dagit`? I would expect a dedicated `DagsterInvalidConfigError` something - are these inner errors caught and turned in to that?

also how come this is not being caught at schema validation time? Doesn't the config schema know the valid keys that are possible?

Reply:
> schema type of a dict type loader is Permissive

ah - thats the root of the problem. We should change it to use the `Map` type that @benpankow  added 
https://github.com/dagster-io/dagster/blame/master/python_modules/dagster/dagster/_config/field_utils.py#L158-L180